### PR TITLE
Fix broken nosetest to pytest fixtures

### PR DESCRIPTION
**Pull request by aweisberg**
Also update README.md with a bunch of corrected/new info.

### CODE REVIEW

**Comment by bereng** [comment]
You may want to mention pyenv as a hint in cases people need to have multiple python installations on the box

File: README.md [hunk 1]
```diff
@@ -7,17 +7,47 @@ localhost.
...
 - A working python installation (tested to work with python 2.7).
-- pyYAML (http://pyyaml.org/ -- `sudo easy_install pyYaml`)
-- six (https://pypi.org/project/six/ -- `sudo easy_install six`)
+- See `requirements.txt` for runtime requirements
+- `mock` and `pytest` for tests
 - ant (http://ant.apache.org/, on Mac OS X, `brew install ant`)
-- psutil (https://pypi.org/project/psutil/)
 - Java (which version depends on the version of Cassandra you plan to use. If
-  unsure, use Java 7 as it is known to work with current versions of Cassandra).
+  unsure, use Java 8 as it is known to work with current versions of Cassandra).
```
Comment (bschoening) [inline README.md:50]: Perhaps reference Java 11 instead here.  Java 8 is on LTS.

File: tests/__init__.py [hunk 2]
```diff
@@ -8,16 +6,3 @@
 
 if sys.version_info < (3, 0):
```
Comment (bschoening) [inline tests/__init__.py:7]: Python < 3.7 is EOL

File: README.md [hunk 3]
```diff
@@ -7,17 +7,47 @@ localhost.
 The goal of ccm and ccmlib is to make it easy to create, manage and destroy a
 small Cassandra cluster on a local box. It is meant for testing a Cassandra cluster.
 
+New to Python development?
+--------------------------
+Python has moved on since CCM started development. `pip` is the new `easy_install`,
+Python 3 is the new 2.7, and virtualenv is strongly recommended for managing multiple Python versions
```
Comment (bschoening) [inline README.md:13]: Suggest: Python 3.7 or later is required and using virtualenv is strongly recommended for managing multiple Python versions

File: README.md (hunk 1 above)
Comment (aweisberg) [reply README.md:50]: Good suggestion. I updated the text. Please LMK what you think.

**Comment by aweisberg** [comment]
@bereng @ekaterinadimitrova2 Hi! I updated this PR based on the feedback. Also added a new gotcha to the readme to help people work around AirPlay also listening on port 7k by default.

File: README.md (hunk 1 above)
Comment (ekaterinadimitrova2) [reply README.md]: I think maybe 8 makes sense until we still maintain as a Community Cassandra versions like 3.x which are on Java 8. I believe we maintain them until next year or next major release.

Comment (aweisberg) [reply README.md]: This comment thread is on an outdated version of the diff. I updated it to reflect that we build on 8 and support 11.

Comment (ekaterinadimitrova2) [reply README.md]: Sounds great! Thank you so much!

**Comment by bereng** [comment]
+1

**Pull request merged**